### PR TITLE
Update AAD baseline to include PIM for Groups

### LINKED ISSUE #376: Update AAD policies to include PIM for Groups
# 💡 Summary #

During Emerald we identified that the existing Azure AD policies that rely on the AAD PIM APIs will need to be augmented to account for the PIM for Groups feature. Currently those baseline policies only include PIM for Roles and therefore are incomplete.

The following AAD policies are impacted by this change:

- 2.11 The Number of Users with the Highest Privilege Roles SHALL Be Limited
- 2.12 Highly Privileged User Accounts SHALL be Cloud-Only
- 2.14 Permanent active role assignments SHALL NOT be allowed for highly privileged roles.
- 2.15 Activation of Highly Privileged Roles SHOULD Require Approval
- 2.16 Highly Privileged Role Assignment and Activation SHALL be Monitored

A detailed impact analysis is provided in a separate comment below. 

## Acceptance criteria ##

- [ ] Update the respective Azure AD baseline policies and implementation steps to account for PIM for Groups
- [ ] Create separate issues to update the respective ScubaGear code

### LINKED ISSUE #797: Update MS.AAD.7.2 Baseline Doc with New Custom Secure Score Calculation
# 💡 Summary #

The current MS.AAD.7.2 policy recommends that the user refer to the Microsoft Identity Secure Score for ensuring that the principal of least privilege is being adequately met. However, per issue #760, the next release will include a custom "Secure Score" calculation for this policy. In order to keep the code and baseline documentation in sync, the following instructions should be updated to reflect the new calculation and to remove the instructions for checking the Microsoft Secure Score value for "Use least privileged administrative roles."

Current baseline:
#### MS.AAD.7.2v1 Instructions

1.  In **Azure Active Directory** select **Security.**

2.  Under **Manage**, select **Identity Secure Score.**

3.  Click the **Columns** button and select all the available columns and click **Apply.**

4.  Review the score for the action named **Use least privileged administrative roles.**

5.  Review the **current score** value and compare it to the **max score**. If the current score is not the maximum value and the status is not **Completed**, you must perform the improvement actions. If that is the case, follow the detailed action steps and check the score again after 48 hours to ensure compliance.

## Motivation and context ##
This documentation update is needed to support the enhancement to the ScubaGear tool which includes a more tangible calculation for the least privilege policy check.

## Implementation notes ##

- This documentation update supports the planned ScubaGear tool enhancement for automatically checking the tenant's least privilege score
- The ScubaGear tool will be able to report the custom Secure Score in the output report
- Users will now be able to view a pass/fail status in the report along with the calculated Secure Score value
- Related issue: #760 

## Acceptance criteria ##

How do we know when this work is done?

- [ ] Documentation accurately reflects the MS.AAD.7.2 code changes

### PR DESCRIPTION
**Pull request by tkol2022**
#  #

Updates the AAD baseline document section 7 privileged user policy instructions to include steps for organizations that are using PIM for Groups. For a description of the changes made refer to https://github.com/cisagov/ScubaGear/issues/376#issuecomment-1901021877.

Closes #376 
Closes #797 

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

**Comment by tkol2022** [comment]
@gdasher @ssatyapal123 Can you please review the revised instruction steps for 7.1? Ignore the paragraphs before 7.1 - I'm still tweaking those.

**Comment by tkol2022** [comment]
@gdasher @ssatyapal123 I finished updating the instructions for 7.1 through 7.5 and it is ready for your review. For a high level description of the changes made refer to https://github.com/cisagov/ScubaGear/issues/376#issuecomment-1901021877.

**Comment by tkol2022** [comment]
pinging reviewers

File: baselines/aad.md [hunk 1]
```diff
@@ -581,42 +583,56 @@ User activation of other highly privileged roles SHOULD trigger an alert.
...
 ### Implementation
 
-The following implementation instructions that reference the Azure AD PIM service will vary if using a third-party PAM system instead. Additionally, the implementation instructions associated with assigning roles to users will be revised in an upcoming release to incorporate functionality provided by the Azure AD [PIM for Groups](https://learn.microsoft.com/en-us/entra/id-governance/privileged-identity-management/concept-pim-for-groups) feature.
+The following implementation instructions that reference the Azure AD PIM service will vary if using a third-party PAM system instead. 
+
+Many of the manual steps described per the instructions in this section are labor intensive. We recommend running the ScubaGear tool instead.
 
 #### MS.AAD.7.1v1 Instructions
 
+When performing these instructions count each distinct username only once. Do not double count. Once you finish counting go to the **Final Count** section.
```
Comment (ssatyapal123) [inline baselines/aad.md:592]:
```suggestion
When performing these instructions count each distinct username only once. Be careful not to double count users listed as both Eligible and Active. Once you finish counting go to the **Final Count** section.
```

**Review by ssatyapal123 — APPROVED** [review]
A couple of minor suggestions but looks good to me otherwise!

**Review by gdasher — COMMENTED** [review]
Should we just direct usage of scubagear (vs. including the manual steps too?)

**Comment by tkol2022** [comment]
Closing due to merge conflicts from files being moved around. New pull request is #872

**Pull request closed without merging**